### PR TITLE
feat(config): add Algolia search configuration to docusaurus.config.js

### LINKED ISSUE #61: [DOCS] Add a search engine, magnifying glass or similar
## Issue Overview

We need something to search the documentation with. 

## Proposed Solutions or Ideas

Docusaurus Search: https://docusaurus.io/docs/search

### PR DESCRIPTION
**Pull request by mimisavage**
# Task name here

Fixes #61

## Types of change

- [ ] Bug
- [x] Enhancement

### CODE REVIEW

**Comment by mimisavage** [comment]
image

**Comment by mimisavage** [comment]
image

**Review by Nadai2010 — APPROVED** [review]
Very good jobs for me, thanks @mimisavage, @metalboyrick can check?

File: docusaurus.config.js
Comment (metalboyrick) [inline docusaurus.config.js:134]: do you know where we deploy this? @Nadai2010 , seems like we might need to extract this to an env. if we do need ENVs we can inform @mimisavage here.

sorry @mimisavage for the delay, normally its not very good practice to hardcode api keys in a repo haha, people can abuse it, let's not rush things

**Review by metalboyrick — CHANGES_REQUESTED** [review]

File: docusaurus.config.js
Comment (metalboyrick) [reply docusaurus.config.js:134]: did you have to generate some keys here ? @mimisavage

Comment (mimisavage) [reply docusaurus.config.js:134]: > did you have to generate some keys here ? @mimisavage

yes i use algolia.
the this the crawler 

image

Comment (mimisavage) [reply docusaurus.config.js:134]: I will add to env and also share the generated api privately via telegram

**Comment by metalboyrick** [comment]
@mimisavage it's the same name as the github.

you should be able to do it here (you can give it any name you like, e.g. `ALGOLIA_API_KEY`). Then use `process.env.ALGOLIA_API_KEY` to access within the app, at the same time creating a new file `.env.example` to put this variable as example

**Comment by mimisavage** [comment]
> @mimisavage it's the same name as the github.
> 
> you should be able to do it here (you can give it any name you like, e.g. `ALGOLIA_API_KEY`). Then use `process.env.ALGOLIA_API_KEY` to access within the app, at the same time creating a new file `.env.example` to put this variable as example

I did this in my recent commit here https://github.com/Scaffold-Stark/ss2-docs/pull/66/commits/f77fe008b50f071380f7530c5b664f2dba687481

**Comment by metalboyrick** [comment]
@mimisavage please wait, I will soon make corresponding changes to the CI so that it will unblock this PR

**Comment by metalboyrick** [comment]
No more comments on my end!

**Review by metalboyrick — APPROVED** [review]
overall looks good, CI PR will be merged after ODHack ends. Seems like we might experience some hiccups while it is not merged yet since server is not set yet. cc @Nadai2010 @0xquantum3labs

**Comment by metalboyrick** [comment]
Thanks for the PR @mimisavage !

**Comment by metalboyrick** [comment]
considering that this PR may break the CI, will be holding it back until other OD PRs are merged.

**Comment by mimisavage** [comment]
> s PR may break the CI, will be holding it back until other OD PRs are merged.

Okay sir. Thank you

**Comment by metalboyrick** [comment]
gm @mimisavage do you have tg that we can connect with?

**Pull request merged**
